### PR TITLE
Update odf-operator direct dependencies

### PR DESCRIPTION
**Pull request by malayparida2000**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
-	k8s.io/apimachinery v0.31.0
-	k8s.io/client-go v0.31.0
+	k8s.io/api v0.31.2
+	k8s.io/apiextensions-apiserver v0.31.1
+	k8s.io/apimachinery v0.31.2
+	k8s.io/client-go v0.31.2
 	sigs.k8s.io/controller-runtime v0.19.0
 )
 
+require github.com/cenkalti/backoff/v4 v4.3.0 // indirect
```
Comment (iamniting) [inline go.mod:25]: Can you add this require in the below section.

**Review by iamniting — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (malayparida2000) [reply go.mod:25]: Yeah seems like it got added automatically, also looking into the go.sum failure.

**Review by iamniting — APPROVED** [review]

**Pull request merged**